rust_src/src/fns.rs

```diff
@@ -274,6 +278,67 @@ pub fn concat(args: &mut [LispObject]) -> LispObject {
     }
 }
 
+/// Return the reversed copy of list, vector, or string SEQ.
+/// See also the function `nreverse', which is used more often.
+#[lisp_fn]
+pub fn reverse(seq: LispObject) -> LispObject {
+    if seq.is_nil() {
+        return Qnil;
+    } else if let Some(cons) = seq.as_cons() {
```

Comment:
This looks like a classic 'fold'. Also, I think `iter_cars` works here. `iter_tails` returns Cons cells and `iter_cars` is `XCAR(cons)` over `iter_tails`. Why the `into` at the end? `new` is already a LispObject right?

```
let new = cons.iter_cars(LispConsEndChecks::on, LispConsCircularChecks::on).fold(Qnil, |cons, elt| {
    LispObject::cons(elt, cons)
});
```

Reply:
Yeah, that looks like it would work. Would remove the need for a `new` as well.

Reply:
If we wanted to be cute we could even write it as `(elt, cons).into()`. I am ok with the explicit LispObject::cons().

Reply:
Yeah, I think the explicitness makes it easier to read.